pkg/http/authorization.go

```diff
@@ -55,7 +54,10 @@ func AuthorizationMiddleware(requireOAuth bool, serverURL string, mcpServer *mcp
 			// Validate the token offline for simple sanity check
 			// Because missing expected audience and expired tokens must be
 			// rejected already.
-			claims, err := validateJWTToken(token, audience)
+			claims, err := ParseJWTClaims(token)
```

Comment:
What about we only have one function that manages parsing, validating, extracting. This function returns scopes and error. Like this;

```go
scopes, err := validateJWTAndGetScopes(token, audience, "") // 3rd parameter is authorization url to check the issuer, if it is set
```

So that we don't need this;

```go
			if err == nil && claims != nil {
				err = claims.Validate(audience)
			}
```